module/zfs/vdev.c

```diff
@@ -1347,6 +1347,10 @@ vdev_open(vdev_t *vd)
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
 			vd->vdev_ashift = ashift;
+		else
+			if (vd->vdev_ashift < ASHIFT_MIN ||
+			    vd->vdev_ashift > ASHIFT_MAX)
```

Comment:
Let's add a new value and then make sure this gets pushed upstream so we stay in sync.    In general the more information we can provide about why the open failed the better.  We're already doubling up on some of the VEDV_AUX errors which can be confusing when debugging.